packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx

```diff
@@ -35,8 +35,7 @@ export const useMultisessionActions = (opts: UseMultisessionActionsParams) => {
 
   const handleManageAccountClicked: React.MouseEventHandler<HTMLButtonElement> = () => {
     if (opts.userProfileMode === 'navigation') {
-      return navigate(opts.userProfileUrl).finally(async () => {
-        await sleep(1000);
+      return navigate(opts.userProfileUrl).finally(() => {
```

Comment:
We're waiting for 1sec for any *external* navigation to happen before closing the popup, as it's rather hard to consistently wait until a browser-navigation takes place, especially across different browsers. 

So we have 2 scenarios we need to handle here: 
- there's an external navigation and the browser will redirect from the host app to accounts (full page reload): we need to show the spinner while the browser is loading and until the page unmounts
- there's an client-side only navigation (eg:  from /home to /user): this we can await so there's no need for the sleep

Reply:
Any ideas on how to check which case we're on?